### PR TITLE
adapted framework construction for multi framework usage

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2507.

### CODE REVIEW

File: service/awsconfig/cluster_framework.go [hunk 1]
```diff
@@ -25,11 +25,7 @@ import (
 	v6cloudconfig "github.com/giantswarm/aws-operator/service/awsconfig/v6/cloudconfig"
 )
 
-const (
-	AWSConfigCleanupFinalizer = "aws-operator.giantswarm.io/custom-object-cleanup"
-)
```
Comment (xh3b4sd) [inline service/awsconfig/cluster_framework.go]: This is left over from the TPR migration. We do not need this at all. Just boyscouting.

File: service/awsconfig/cluster_framework.go [hunk 2]
```diff
@@ -25,11 +25,7 @@ import (
...
-
-type FrameworkConfig struct {
+type ClusterFrameworkConfig struct {
```
Comment (xh3b4sd) [inline service/awsconfig/cluster_framework.go]: We want to have multiple frameworks, so the naming has to be aligned a bit.

File: service/awsconfig/cluster_framework_test.go [hunk 3]
```diff
@@ -9,8 +9,8 @@ import (
 	kubernetesfake "k8s.io/client-go/kubernetes/fake"
 )
 
-func newTestFrameworkConfig() FrameworkConfig {
-	return FrameworkConfig{
+func newTestClusterFrameworkConfig() ClusterFrameworkConfig {
```
Comment (xh3b4sd) [inline service/awsconfig/cluster_framework_test.go]: Just fixing existing tests.

File: service/awsconfig/drainer_framework.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+type DrainerFrameworkConfig struct {
+}
+
+func NewDrainerFramework(config DrainerFrameworkConfig) (*framework.Framework, error) {
+	return nil, nil
+}
+
+func newDrainerResourceRouter(config DrainerFrameworkConfig) (*framework.ResourceRouter, error) {
+	return nil, nil
+}
```
Comment (xh3b4sd) [inline service/awsconfig/drainer_framework.go]: Here will be the new framework magic. Stay tuned for the following PRs.

File: service/service.go [hunk 5]
```diff
@@ -62,10 +62,10 @@ func DefaultConfig() Config {
 
 type Service struct {
 	// Dependencies.
-	Alerter            *alerter.Service
-	AWSConfigFramework *framework.Framework
-	Healthz            *healthz.Service
-	Version            *version.Service
+	Alerter          *alerter.Service
+	ClusterFramework *framework.Framework
```
Comment (xh3b4sd) [inline service/service.go]: We will register the new framework once it is here. For now just renaming.

File: service/awsconfig/drainer_framework.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package awsconfig
+
+import (
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+type DrainerFrameworkConfig struct {
+}
```
Comment (kopiczko) [inline service/awsconfig/drainer_framework.go]: My eyes would be happier if it was called `DrainFramework`, as we don't have `ClustererFramework`. It isn't important though.

Comment (xh3b4sd) [reply service/awsconfig/drainer_framework.go]: It is the cluster and the drainer. Both are nomen. Drain is a verb. To drain. To cluster is not what the term cluster is about.

Comment (xh3b4sd) [reply service/awsconfig/drainer_framework.go]: It is also that cluster and drainer are words having the same length, which makes my eyes very happy. 

```
cluster
drainer
```

**Pull request merged**